src/ComponentBase.ts

```diff
@@ -120,8 +120,8 @@ export abstract class ComponentBase<P extends React.Props<any>, S extends Object
     componentWillReceiveProps(nextProps: Readonly<P>, nextContext: any): void {
         _.forEach(this._handledSubscriptions, (subscription: StoreSubscriptionInternal<S>) => {
             if (subscription.keyPropertyName) {
-                let curVal = _.get<string>(this.props, subscription.keyPropertyName);
-                let nextVal = _.get<string>(nextProps, subscription.keyPropertyName);
+                let curVal = _.get<P, any>(this.props, subscription.keyPropertyName);
```

Comment:
the key only works as a string -- shouldn't be any

Reply:
I've reverted `@types/lodash`, because in [new version of types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/index.d.ts#L13343) were drastically changed, and need find an appropriate way how to we can use them in our case.  

`_.get<P, string>(this.props, subscription.keyPropertyName)` will now work, because `TKey extends keyof TObject`. 

I need some time to investigate this issue. I applied changes from `master` and pinned `@types/lodash` to the version which is using in `master`.